### PR TITLE
ref(rules): Bucket max rules into slow and fast

### LINKED ISSUE #51422: Ability to exceed 100 alert issues per project
### Problem Statement

For multiple teams sharing a project, the total number of alerts per project can easily exceed the 100 alert limit.  

### Solution Brainstorm

Differentiating and having a higher limit for the rules that are not as resource intensive, for example alert rules that query snuba can be expensive

### Product Area

Alerts

### PR DESCRIPTION
**Pull request by ceorourke**
Allow for 100 "slow" rules (these are more expensive to process) and 200 "fast" issue alert rules per project.

Closes #51422

### CODE REVIEW

File: src/sentry/api/endpoints/project_rules.py [hunk 1]
```diff
@@ -61,83 +62,94 @@ def post(self, request: Request, project) -> Response:
...
+        for rule in rules:
+            for condition in rule.data["conditions"]:
+                for slow_conditions in SLOW_CONDITION_MATCHES:
+                    if slow_conditions in condition["id"]:
+                        slow_rules += 1
+                    break
+
+        if slow_rules >= settings.MAX_SLOW_CONDITION_ALERTS:
+            return Response(
+                f"You may not exceed {settings.MAX_SLOW_CONDITION_ALERTS} 'slow' rules per project.",
```
Comment (ceorourke) [inline src/sentry/api/endpoints/project_rules.py:89]: Users aren't going to know what this means, so I was thinking I'd add a blurb in the docs somewhere, then go back and add a link to the docs once it exists.

File: src/sentry/api/endpoints/project_rules.py [hunk 2]
```diff
@@ -61,83 +62,94 @@ def post(self, request: Request, project) -> Response:
...
+        conditions = data.get("conditions", [])
+        if "filters" in data:
+            conditions.extend(data["filters"])
+
+        rules = Rule.objects.filter(project=project, status=RuleStatus.ACTIVE)
+        slow_rules = 0
+        for rule in rules:
+            for condition in rule.data["conditions"]:
+                for slow_conditions in SLOW_CONDITION_MATCHES:
+                    if slow_conditions in condition["id"]:
```
Comment (wedamija) [inline src/sentry/api/endpoints/project_rules.py:83]: Might be worth breaking this out into a `is_condition_slow` function that we can share with the rule processor

File: src/sentry/conf/server.py [hunk 3]
```diff
@@ -3374,7 +3374,8 @@ def build_cdc_postgres_init_db_volume(settings: Any) -> dict[str, dict[str, str]
 # determines if we enable analytics or not
 ENABLE_ANALYTICS = False
 
-MAX_ISSUE_ALERTS_PER_PROJECT = 100
+MAX_SLOW_CONDITION_ALERTS = 100
+MAX_FAST_CONDITION_ALERTS = 200
```
Comment (wedamija) [inline src/sentry/conf/server.py:3378]: `*_ISSUE_ALERTS` just to make sure we differentiate from metric alerts

File: src/sentry/rules/processor.py [hunk 4]
```diff
@@ -187,11 +194,7 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
                 filter_list.append(rule_cond)
 
         # Sort `condition_list` so that most expensive conditions run last.
-        condition_list.sort(
-            key=lambda condition: any(
-                condition_match in condition["id"] for condition_match in SLOW_CONDITION_MATCHES
-            )
-        )
+        condition_list.sort(key=lambda condition: any([is_condition_slow(condition)]))
```
Comment (wedamija) [inline src/sentry/rules/processor.py:197]: Nit: I think we can remove the `any([])` now and just use the return value

**Review by wedamija — APPROVED** [review]

**Pull request merged**